### DISCUSSION TITLE
Exposing collector prometheus metrics

### DISCUSSION THREAD

**Discussion by jcantrill** [vectordotdev/vector#7486] [Q&A] [answered]
Is there any documentation on how to expose prometheus metrics for the collector itself, not as an intermediary to pull/push metrics from source to destination?

**Comment by jszwedko** [comment]
Hi @jcantrill . Do you mean metrics like those collected by [`internal_metrics`](https://vector.dev/docs/reference/configuration/sources/internal_metrics/)? You can send those to a `prometheus_exporter` sink.

**Reply by jcantrill** [reply]
Yes, those where the ones I couldnt find.  Does vector only support a push model?  Can it also support pull for internal metrics?

**Reply by jszwedko** [reply, marked as answer]
The `prometheus_exporter` sink is pull-based. It binds to an address and exposes the metrics at `/metrics` in the prometheus exposition format so that they can be scraped. Is that what you are looking for?
